plugin/src/main/java/io/jenkins/plugins/analysis/core/portlets/PullRequestMonitoringPortlet.java

```diff
@@ -0,0 +1,130 @@
+package io.jenkins.plugins.analysis.core.portlets;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.google.gson.JsonObject;
+
+import edu.hm.hafner.analysis.Severity;
+
+import org.kohsuke.stapler.bind.JavaScriptMethod;
+import hudson.Extension;
+import hudson.model.Run;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.model.ResultAction;
+import io.jenkins.plugins.monitoring.MonitorPortlet;
+import io.jenkins.plugins.monitoring.MonitorPortletFactory;
+
+/**
+ * A portlet that can be used for the pull-request-monitoring dashboard
+ * (https://github.com/jenkinsci/pull-request-monitoring-plugin).
+ * It renders an interactive sunburst diagram for each {@link ResultAction},
+ * which is registered at the current users {@link Run}.
+ *
+ * @author Simon Symhoven
+ */
+public class PullRequestMonitoringPortlet implements MonitorPortlet {
+
+    private final ResultAction action;
+
+    /**
+     * Creates a new {@link PullRequestMonitoringPortlet}.
+     *
+     * @param action
+     *              the corresponding {@link ResultAction}.
+     */
+    public PullRequestMonitoringPortlet(final ResultAction action) {
+        this.action = action;
+    }
+
+    @Override
+    public String getTitle() {
+        return action.getLabelProvider().getName();
+    }
+
+    @Override
+    @JavaScriptMethod
+    public String getId() {
+        return action.getResult().getId();
```

Comment:
I wonder if this ID should also include "warnings", to reduce the risk of other plugins defining portlet IDs that clash with IDs that Warnings NG copies from AnalysisModelParserDescriptor. However, perhaps the URL would then clash anyway.

Reply:
Good point. In general it makes sense, if the artifact id of the plugin is added as a suffix.